### DISCUSSION TITLE
MS Test (trx files) with Allure

### DISCUSSION THREAD

**Discussion by krmalive** [allure-framework/allure2#1661] [General Discussion]
Is there any documentation available on how to use ms test with allure?
In the [current documentation](https://docs.qameta.io/allure/#_mstest), it says about **AllureMSTestAdapter**, however on the[ GitHub repository](https://github.com/allure-framework/allure-mstest) the solution is mentioned as **Deprecated**
It would be great if someone can help on this.

Thanks in advance,
Karthik

**Comment by baev** [comment]
`TRX` files are now supported natively by both Allure Report & Allure TestOps. Simply specify the directory with `TRX` files to generate the report

`AllureMSTestAdapter ` is deprecated as it only was used to convert TRX files to Allure json files.

**Reply by krmalive** [reply]
Thank you, It worked for me.
Appreciate it

**Reply by baev** [reply]
awesome! Could you please mark it as an answer?

**Comment by krmalive** [comment]
I have an addon question too,

I tried Allure Nunit as well, and  when I generate report using c# code, I get the allure results in json files with container and result as end file name.
<img width="603" alt="NUnit" src="https://user-images.githubusercontent.com/105714930/170054435-8b3cc607-a0c3-4bd6-b79b-c2bdbe21f877.PNG">

However, when I generate report directly from trx file using "allure generate" command, I get the results in completely different format.
![image](https://user-images.githubusercontent.com/105714930/170054344-789c159b-c07a-4bd0-8781-aa7390a71bcb.png)

I checked the 'data' folder as well, but the format is different.
![image](https://user-images.githubusercontent.com/105714930/170054779-70232fa1-dc2d-420d-88ac-922f3428f0d0.png)


How do I get the result like I get with NUnit or Specflow, I mean to say, is there any way to specify annotations while generating reports from trx files?

Thank in advance,
Karthik
